src/prefect/cli/deployment.py

```diff
@@ -273,12 +273,16 @@ async def apply(
             exit_with_error(f"'{path!s}' did not conform to deployment spec: {exc!r}")
 
         if upload:
-            file_count = await deployment.upload_to_storage()
-            if file_count:
-                app.console.print(
-                    f"Successfully uploaded {file_count} files to {deployment.location}",
-                    style="green",
-                )
+            if (
+                deployment.storage
+                and "put-directory" in deployment.storage.get_block_capabilities()
```

Comment:
👏 I love how the capabilities worked out from the calling side.  We may want to add a `Block.has_capability(self, capability: str) -> bool` method to make these read even more clearly, but it's already pretty good.

Reply:
Also, you didn't just enable Github deployments, you enabled using any read-only storage block :bow:

Reply:
Agreed, this `x in deployment.storage.get_block_capabilities()` pattern is common enough at this point that it probably warrants a helper.